File: engine/model/src/main/scala/tools/packet/partyoperation/PartyInvite.scala
package tools.packet.partyoperation

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class PartyInvite(private var _partyId: Int, private var _fromCharacterName: String) extends PacketInput {
  def partyId: Int = _partyId

  def fromCharacterName: String = _fromCharacterName

  override def opcode(): SendOpcode = SendOpcode.PARTY_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/party/PartyInvite.java
package tools.packet.party;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record PartyInvite(Integer partyId, String fromCharacterName) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.PARTY_OPERATION;
   }
}